pilot/pkg/model/xds_cache.go

```diff
@@ -185,19 +188,22 @@ func (l *lruCache) recordDependentConfigSize() {
 }
 
 // This is the callback passed to LRU, it will be called whenever a key is removed.
-func (l *lruCache) evict(k interface{}, v interface{}) {
+func (l *lruCache) onEvict(k interface{}, v interface{}) {
 	if features.EnableXDSCacheMetrics {
-		xdsCacheEvictions.Increment()
-	}
-
-	if !l.evicted {
```

Comment:
yes, this seems like a clear leak, the tests I had on another PR should've faild with this change

Reply:
i see that https://github.com/istio/istio/pull/39688 has been reverted, which changed the behaviour of the cache and made my tests invalid, I could now probably merge them.